gslb/gslbutils/gdputils.go

```diff
@@ -363,6 +414,54 @@ func isSyncTypeChanged(new, old *gdpv1alpha2.GlobalDeploymentPolicy) []string {
 	return clustersToBeSynced
 }
 
+func isHmRefsChanged(new, old *gdpv1alpha2.GlobalDeploymentPolicy) bool {
+	if len(old.Spec.HealthMonitorRefs) != len(new.Spec.HealthMonitorRefs) {
+		return true
+	}
+	oldHmRefs := make(map[string]struct{})
+	for _, hmRef := range old.Spec.HealthMonitorRefs {
+		oldHmRefs[hmRef] = struct{}{}
+	}
+	for _, hmRef := range new.Spec.HealthMonitorRefs {
+		if _, exists := oldHmRefs[hmRef]; !exists {
+			return true
+		}
+	}
+	return false
+}
+
+func isSitePersistenceChanged(new, old *gdpv1alpha2.GlobalDeploymentPolicy) bool {
+	newSp := new.Spec.SitePersistence
+	oldSp := old.Spec.SitePersistence
+	if (newSp.Enabled == oldSp.Enabled) && (newSp.Enabled == true) {
+		// if both are true, we check the profile ref value
+		if newSp.ProfileRef != oldSp.ProfileRef {
+			return true
+		}
+		return false
+	} else if newSp.Enabled != oldSp.Enabled {
+		// else, check for any change in the enabled property
+		return true
+	}
+	return false
+}
+
+func isTTLChanged(new, old *gdpv1alpha2.GlobalDeploymentPolicy) bool {
+	if new.Spec.TTL == nil && old.Spec.TTL != nil {
+		return true
+	} else if new.Spec.TTL != nil && old.Spec.TTL == nil {
+		return true
+	} else if new.Spec.TTL != nil && old.Spec.TTL != nil && *new.Spec.TTL != *old.Spec.TTL {
```

Comment:
why is there a pointer comparison here?

Reply:
I am not sure what you mean. `*new.Spec.TTL` is the value of the TTL field in the new GDP object and similar thing for the old object. So we are just comparing the values.